tests/repository_simulator.py

```diff
@@ -234,35 +246,35 @@ def _fetch_metadata(
             # return a version previously serialized in publish_root()
             if version is None or version > len(self.signed_roots):
                 raise FetcherHTTPError(f"Unknown root version {version}", 404)
```

Comment:
Shall we log something before raising the ``FetcherHTTPError``? 
If it makes sense would be good to apply also to lines 263, 265

Reply:
I don't think this is needed here: the repository (or the FetcherInterface implementation to be exact) is saying it does not have this metadata, that's it (we also have zero other details to log)